src/bootstrap/build/sanity.rs

```diff
@@ -78,6 +79,24 @@ pub fn check(build: &mut Build) {
             panic!("the iOS target is only supported on OSX");
         }
 
+        if cfg!(windows) {
```

Comment:
This also seems relevant for all platforms (not just Windows), so perhaps this could create a variable like `ldpath` to handle the Unix/OSX cases?